### ISSUE TITLE
Finalize the list of files which require modifications.

### ISSUE THREAD

**Issue by theSaurabhMhatre** [theSaurabhMhatre/expertiza#4]
Scrutinize the tentative files list, set aside the files which we will have to modify and explain in brief the changes we will be making to those files in wiki.

**Comment by theSaurabhMhatre** [comment]
Specified additional details for the methods listed in the wiki. Will require further analysis to specify the changes required in depth.

**Comment by theSaurabhMhatre** [comment]
Scrutinized all the identified files, and it seems that the questions weights are being taken into consideration. A summary of the findings is as below:

1. When adding a new question to a questionnaire, the update method gets called which redirects the flow to add_new_questions method which sets the weight of scored questions to 1 by default when such a question is created for the first time. We could not find a way to associate a weight with a scored question when creating it for the first time.

2. After creating such a scored question, we changed the weight and saved the questionnaire, which resulted in the update method being called. This time we observed that params[:question] contained all of the existing questions with updated attributes if any (the updated weight values also get recorded here), which are then used to update the weights and any other attributes of the corresponding questions in the database. This is how the weights get updated, we also verified the same after checking the corresponding weights in the database.

3. Also, all such weights associated with scored questions in a questionnaire are considered when calculating the grades as is evident from the get_total_score method in answer.rb. This method calculates the score for a questionnaire by considering one response at a time which is then aggregated by the caller of this method, compute_scores in the same file. 

To calculate the scores for scored questions, the following query is being used:

questionnaireData = ScoreView.find_by_sql ["SELECT q1_max_question_score ,SUM(question_weight) as sum_of_weights,SUM(question_weight * s_score) as weighted_score FROM score_views WHERE type in('Criterion', 'Scale') AND q1_id = ? AND s_response_id = ?", @questions[0].questionnaire_id, @response.id] 

We can see that the above query multiplies the score with weights associated with questions. We were also able to verify this by creating a questionnaire and associating different weights with the questions in it. This led to the score calculation for that questionnaire to change as per the weights, which we were also able to verify manually.
